dali/kernels/common/split_shape.h

```diff
@@ -59,6 +63,53 @@ void split_shape(SplitFactor& split_factor, const Shape& in_shape, int min_nbloc
   }
 }
 
+/**
+ * @brief returns the dimension index with a split factor > 1
+ */
+template <typename SplitFactor>
+int LastSplitDim(const SplitFactor& split_factor) {
+  int last_split_dim = -1;
+  int ndim = dali::size(split_factor);
+  for (int d = ndim - 1; d >= 0; d--) {
+    if (split_factor[d] > 1) {
+      last_split_dim = d;
+      break;
+    }
+  }
+  return last_split_dim;
+}
+
+/**
+ * @brief Schedule work on a per-block basis
+ */
+template <int ndim, typename SplitFactor, typename ScheduleBlockFunc>
```

Comment:
I don't think it has much to do with slip_shape.

Reply:
Can you provide a description how it works and what is that for?

Reply:
Initially the function did split the shape, but then it evolved into one that doesn't store the blocks but instead calls a function with the subregions as parameters. It could be renamed or the file could be renamed.
Proposed alternative:
```
template <typename OnBlockFunc>
void ScheduleBlocks(const TensorShape<ndim> &start, const TensorShape<ndim> &end,
                    const SplitFactor &split_factor, int d, int max_split_dim,
                    OnBlockFunc &&on_block) {
```

Reply:
Rename is fine. Still I would appreciate the docs here.

Reply:
I'll rename and document in the original PR